pkg/deploy/kubernetes_image_puller.go

```diff
@@ -44,6 +44,20 @@ type ImageAndName struct {
 // a KubernetesImagePuller CR.  Add a finalizer to the CheCluster CR.  If false, remove the KubernetesImagePuller CR, uninstall the operator, and remove the finalizer.
 func ReconcileImagePuller(ctx *DeployContext) (reconcile.Result, error) {
 
+	if !ctx.CheCluster.ObjectMeta.DeletionTimestamp.IsZero() {
```

Comment:
Could we avoid code duplication with: 
https://github.com/eclipse-che/che-operator/blob/main/pkg/deploy/kubernetes_image_puller.go#L173-L194